### PR TITLE
Enum support refactor suggestion

### PR DESCRIPTION
**Pull request by georgyangelov**
Thought to try it out instead of commenting. Fixes the issue of having mixed enums getting defined incorrectly as strings. See this change:

Previous: `expect(values).toEqual(['42', 'test']);`, new: `expect(values).toEqual([42, 'test']);`
and also mixed enums should be an error from OpenAPI generation standpoint because we can't support both without doing some `anyOf` schenanigans which I didn't want to think about.

### CODE REVIEW

File: spec/simple.spec.ts [hunk 1]
```diff
@@ -199,6 +199,63 @@ describe('Simple', () => {
...
+    expectSchema([nativeEnumSchema], {
+      NativeEnum: {
+        type: 'string',
+        description: 'A native string enum in zod',
+        enum: ['a', 'b', 'c'],
+      },
+    });
+  });
+
+  it('does not support mixed native enums', () => {
```
Comment (AGalabov) [inline spec/simple.spec.ts:223]: Maybe a describe for enums could help here. Not sure if you already have this in the other PR

File: src/openapi-generator.ts [hunk 2]
```diff
@@ -610,14 +611,26 @@ export class OpenAPIGenerator {
...
+
+      if (type === 'mixed') {
+        // enum Test {
+        //   A = 42,
+        //   B = 'test',
+        // }
+        //
+        // const result = z.nativeEnum(Test).parse('42');
+        //
+        // This is an error, so we can't just say it's a 'string'
```
Comment (AGalabov) [inline src/openapi-generator.ts:624]: Nice

File: src/lib/enum-info.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  const numericCount = values.filter(_ => typeof _ === 'number').length;
+
+  const type =
+    numericCount === 0
+      ? ('string' as const)
+      : numericCount === values.length
+      ? ('numeric' as const)
+      : ('mixed' as const);
+
+  return { values: values, type };
```
Comment (AGalabov) [inline src/lib/enum-info.ts:23]: Just `{ values, type }`

File: spec/simple.spec.ts [hunk 4]
```diff
@@ -199,6 +199,63 @@ describe('Simple', () => {
     });
   });
 
+  it('supports native string enums', () => {
```
[the remark this answers is not in the document]
Comment (georgyangelov) [reply spec/simple.spec.ts]: How did I not see that

File: spec/simple.spec.ts (hunk 1 above)
Comment (georgyangelov) [reply spec/simple.spec.ts:202]: I have that in the other PR

File: src/lib/enum-info.ts (hunk 3 above)
Comment (georgyangelov) [reply src/lib/enum-info.ts]: Done

**Review by AGalabov — APPROVED** [review]

**Pull request merged**